satpy/readers/abi_l1b.py

```diff
@@ -81,8 +81,9 @@ def get_dataset(self, key, info):
         # although we could compute these, we'd have to update in calibration
         res.attrs.pop('valid_range', None)
         # add in information from the filename that may be useful to the user
-        for attr in ('observation_type', 'scene_abbr', 'scan_mode', 'platform_shortname'):
-            res.attrs[attr] = self.filename_info[attr]
+        for attr in ('observation_type', 'scene_abbr', 'scan_mode', 'platform_shortname', 'suffix'):
```

Comment:
I know this is not new, but I don't really like having to add to this everytime there is a new item popping up in the file patterns. Could these items to be passed to the user be defined in the yaml file instead?

Reply:
I was thinking the former, in the file type definition, but maybe the reader section is more appropriate

Reply:
@mraspaud I was going to try to implement this, but realized the problem with defining this list of attrs in the reader (the more logical place) makes the code difficult because the file handler's don't have access to the reader kwargs...oh unless I modify the ABI Base file handler and have it expect this kwarg. :thinking: Will look tomorrow, but let me know what you think.

Reply:
I think having them in the filetype info would be ok, because you might want to have different attributes for different filetypes.

Reply:
Yeah but then I have 16 repeat blocks of YAML **plus** the parameters are typically per "style" of file pattern and not necessarily file type. I'd just like to avoid this "clean up". It is beginning to not be clean at all. Let me know how strongly you care.

Reply:
I care about how clean the solution is more than where this info is :) so it's up to you!

Reply:
How about this: If this list of attributes changes again then I promise I will find a clean solution for this where the values are put somewhere more configure-y (YAML). Honestly, we could probably just copy all the filename parameters if they aren't already defined earlier in `get_dataset`. I'm going to merge this.

Reply:
ok, it's up to you. I'll take you on that promise though :D

Reply:
The abi l1b reader (at least the `get_dataset` method) is gross enough as it is. It needs to be refactored but this isn't the PR or the time for it. I know that technically breaks the "law of refactoring" but I didn't get enough sleep last night to care :wink: